apis/leds/src/lib.rs

```diff
@@ -0,0 +1,60 @@
+#![no_std]
+
+use libtock_platform::Syscalls;
+
+/// The LEDs driver
+///
+/// # Example
+/// ```ignore
+/// use libtock2::Leds;
+///
+/// // Turn on led 0
+/// Leds::on(0);
+/// ```
+pub struct Leds<S: Syscalls>(S);
+
+impl<S: Syscalls> Leds<S> {
+    /// Run a check against the leds capsule to ensure it is present.
+    ///
+    /// Returns `true` if the driver was present. This does not necessarily mean
+    /// that the driver is working, as it may still fail to allocate grant
+    /// memory.
+    #[inline(always)]
+    pub fn driver_check() -> bool {
+        S::command(DRIVER_ID, DRIVER_CHECK, 0, 0).is_success_u32()
+    }
+
+    #[inline(always)]
+    pub fn count() -> usize {
+        S::command(DRIVER_ID, LEDS_COUNT, 0, 0)
+            .get_success_u32()
+            .unwrap_or_default() as usize
+    }
```

Comment:
Should probably merge `driver_check` and `count` into a single function that returns `Option<u32>`, with `None` indicating the driver is missing.

Reply:
I assumed `driver_check` should present with all drivers. I changed `count` to return `Option<usize>`.

Reply:
I think `Option<u32>` is better. `Option<usize>` indicates that you want apps to be able to use LED numbers greater than `u32::MAX` in 64-bit environments, such as the unit test environment. I don't think that unit tests should be designed to use LED counts that a real Tock system is unable to achieve.